### PR TITLE
Implements Elevator and Teleporter mechanic while riding a horse.

### PR DESCRIPTION
**Pull request by Xericore**
This also fixes Elevator when smooth-movement is true (https://dev.enginehub.org/youtrack/issue/CRAFTBOOK-3464).

### CODE REVIEW

**Comment by Xericore** [comment]
Looks like I was a bit too optimistic with the smooth-movement fixes. I created more test cases now with blocks like fences, glass panes, etc. Turns out the player can get stuck in the elevator, because only checking the block above the player is not enough. I have an idea how to solve that, but it would definitely take some time to implement it.

File: src/main/java/com/sk89q/craftbook/mechanics/Elevator.java [hunk 1]
```diff
@@ -22,25 +22,21 @@
 import com.sk89q.craftbook.bukkit.BukkitCraftBookPlayer;
 import com.sk89q.craftbook.bukkit.CraftBookPlugin;
 import com.sk89q.craftbook.bukkit.util.CraftBookBukkitUtil;
-import com.sk89q.craftbook.util.EventUtil;
-import com.sk89q.craftbook.util.LocationUtil;
-import com.sk89q.craftbook.util.ProtectionUtil;
-import com.sk89q.craftbook.util.RegexUtil;
-import com.sk89q.craftbook.util.SignUtil;
+import com.sk89q.craftbook.util.*;
```
Comment (me4502) [inline src/main/java/com/sk89q/craftbook/mechanics/Elevator.java]: Would you be able to please revert these import changes?

Comment (Xericore) [reply src/main/java/com/sk89q/craftbook/mechanics/Elevator.java]: Sure, done in 8406fdc.

**Comment by me4502** [comment]
Nice, thanks

**Comment by Xericore** [comment]
Thanks you for merging it. I'll have a look at improving the smooth-movement mechanic somewhen in the future. I really like it, but it needs to have a more sophisticated collision avoidance mechnism for function robustly. I'll let you know when I have something better. Best regards.

**Pull request merged**